build/net/src/lib.rs

```diff
@@ -45,7 +58,35 @@ pub struct TaskNote {
 }
 
 pub fn load_net_config() -> Result<NetConfig, Box<dyn std::error::Error>> {
-    Ok(build_util::config::<GlobalConfig>()?.net)
+    let cfg = build_util::config::<GlobalConfig>()?.net;
+
+    match (cfg!(feature = "vlan"), cfg.vlan.is_some()) {
+        (true, false) => {
+            panic!("VLAN feature is enabled, but vlan is missing from config")
+        }
+        (false, true) => {
+            panic!("VLAN feature is disabled, but vlan is present in config")
+        }
+        _ => (),
+    }
+
+    Ok(cfg)
+}
+
+pub fn generate_vlan_consts(
+    config: &NetConfig,
+    mut out: impl std::io::Write,
+) -> Result<(), std::io::Error> {
+    let vlan = config.vlan.unwrap();
+    let end = vlan.start + vlan.count;
+    writeln!(
+        out,
+        "
+pub const VLAN_RANGE: core::ops::Range<u16> = {:#x}..{:#x};
```

Comment:
Would we ever want to be able to include `0xFFFF` here? If so, you may wish to use a `RangeInclusive`.

Reply:
Nope, the VID is actually only 12 bits. I just added checks + masking to ignore the upper 4 bits.